api/src/pcapi/core/educational/api/shared.py

```diff
@@ -66,6 +65,10 @@ def _update_collective_booking_pending(expired_booking: models.CollectiveBooking
 
 
 def _should_update_collective_booking_pending(
-    booking: models.CollectiveBooking | None, booking_limit: datetime.datetime
+    booking: models.CollectiveBooking, booking_limit: datetime.datetime
 ) -> bool:
-    return booking is not None and booking.is_expired and booking_limit > date_utils.get_naive_utc_now()
+    # if the input date has a timezone (resp. does not have one), we need to compare it with an aware datetime (resp. a naive datetime)
+    now = (
+        date_utils.get_naive_utc_now() if booking_limit.tzinfo is None else datetime.datetime.now(datetime.timezone.utc)
```

Comment:
```suggestion
        date_utils.get_naive_utc_now() if booking_limit.tzinfo is None else datetime.datetime.now(datetime.UTC)
```

[Doc](https://docs.python.org/3/library/datetime.html#datetime.UTC) 

Mais c'est du détail, fais comme tu le sens :)